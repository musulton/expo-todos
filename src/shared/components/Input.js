import React from "react";
import {TextInput, StyleSheet} from "react-native";

const Input = ({value, onChangeValue}) => {
    return (
        <TextInput
            value={value}
            onChangeText={onChangeValue}
            placeholder={"What needs to be done?"}
            style={styles.input}
            placeholderTextColor={"#CACACA"}
            selectionColor={"#666666"}
        />
    )
}

export default Input;

const styles = StyleSheet.create({
    input: {
        padding: 10,
        borderWidth: 1,
        borderColor: "black",
    }
})
