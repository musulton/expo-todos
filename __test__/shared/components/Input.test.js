import {fireEvent, render, screen} from "@testing-library/react-native";

import Input from "../../../src/shared/components/Input";

describe("Input", () => {
    const mockProps = {
        value: "Value",
        onChangeValue: jest.fn(),
        placeholder: "Placeholder"
    }

    test("should render correctly", () => {
        render(<Input {...mockProps} />)

        expect(screen.toJSON()).toMatchSnapshot()
    })

    test("should call onChangeValue when change input text", () => {
        const value = "new value"

        render(<Input {...mockProps} />)
        fireEvent.changeText(screen.getByPlaceholderText(mockProps.placeholder), value)

        expect(mockProps.onChangeValue).toHaveBeenCalledWith(value)
    })
})
