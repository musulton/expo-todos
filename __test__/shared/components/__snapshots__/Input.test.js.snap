// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<TextInput
  onChangeText={[MockFunction]}
  placeholder="Placeholder"
  placeholderTextColor="#CACACA"
  selectionColor="#666666"
  style={
    {
      "borderColor": "black",
      "borderWidth": 1,
      "marginVertical": 3,
      "padding": 10,
    }
  }
  value="Value"
/>
`;
